libsolidity/codegen/ir/IRLValue.h

```diff
@@ -107,6 +108,25 @@ class IRStorageItem: public IRLValue
 	boost::variant<std::string, unsigned> const m_offset;
 };
 
+/**
+ * Reference to the "length" member of a dynamically-sized array. This is an LValue with special
+ * semantics since assignments to it might reduce its length and thus arrays members have to be
+ * deleted.
+ */
+class IRStorageArrayLength: public IRLValue
+{
+public:
+	/// Constructs the LValue, assumes that the reference to the array head is already on the stack.
+	IRStorageArrayLength(IRGenerationContext& _context, std::string const _slot, Type const* _type);
```

Comment:
```suggestion
	IRStorageArrayLength(IRGenerationContext& _context, std::string const& _slot, Type const* _type);
```

Reply:
I removed the `const` instead so that the `move()` in the c'tor works